pkg/services/ngalert/api/api_testing.go

```diff
@@ -56,8 +58,27 @@ func (srv TestingApiSrv) RouteTestGrafanaRuleConfig(c *contextmodel.ReqContext,
 
 	conditionEval, err := srv.evaluator.Create(ctx, evalCond)
 	if err != nil {
-		return ErrResp(http.StatusBadRequest, err, "invalid condition")
-	}
+		return ErrResp(http.StatusInternalServerError, err, "Failed to evaluate queries")
+	}
+
+	cfg := state.ManagerCfg{
+		Metrics:                 nil,
+		ExternalURL:             srv.appUrl,
+		InstanceStore:           nil,
+		Images:                  &backtesting.NoopImageService{},
+		Clock:                   clock.New(),
+		Historian:               nil,
+		MaxStateSaveConcurrency: 1,
+	}
+	manager := state.NewManager(cfg)
+	includeFolder := !srv.cfg.ReservedLabels.IsReservedLabelDisabled(models.FolderTitleLabel)
+	transitions := manager.ProcessEvalResults(
+		c.Req.Context(),
+		now,
+		rule,
+		results,
+		state.GetRuleExtraLabels(rule, body.NamespaceTitle, includeFolder),
+	)
```

Comment:
I think this should be removed at all as it is not used by this API before 10.1.x

Reply:
Yep, I forgot to discard these changes, thanks!